src/main/java/com/google/api/codegen/discovery/transformer/java/JavaSampleTypeNameConverter.java

```diff
@@ -62,29 +61,26 @@
   private final TypeNameConverter typeNameConverter;
   private final String packagePrefix;
 
-  public JavaSampleTypeNameConverter(String packagePrefix) {
+  public JavaSampleTypeNameConverter(String apiName, String apiVersion, boolean versionModule) {
     this.typeNameConverter = new JavaTypeTable("");
-    this.packagePrefix = packagePrefix;
+    StringBuilder stringBuilder = new StringBuilder("com.google.api.services.").append(apiName);
+    if (versionModule) {
+      stringBuilder.append('.');
+      stringBuilder.append(apiVersion);
+    }
+    this.packagePrefix = stringBuilder.toString();
   }
 
   @Override
   public TypeName getServiceTypeName(String apiTypeName) {
     return typeNameConverter.getTypeName(Joiner.on('.').join(packagePrefix, apiTypeName));
   }
 
-  /** Returns packagePrefix with subpackage appended if not empty. */
-  private String packagePrefix(String subpackage) {
-    if (!Strings.isNullOrEmpty(subpackage)) {
-      return new StringBuilder(packagePrefix).append('.').append(subpackage).toString();
-    }
-    return packagePrefix;
-  }
-
   @Override
   public TypeName getRequestTypeName(String apiTypeName, TypeInfo typeInfo) {
```

Comment:
If I understand correctly you mean calling directly into `getTypeName` from `getRequestTypeName` so we can remove the bit of redundancy?

It's tricky because they aren't constructed the same way...

model types don't have `apiTypeName` as one of their parts, and also follow different rules  because the request types must be partially qualified.